### DISCUSSION TITLE
Advanced Notifications

### DISCUSSION THREAD

**Discussion by hariria** [tauri-apps/tauri#6017] [Ideas]
At the moment Tauri supports the following for [notification options](https://tauri.app/v1/api/js/notification#options)
- body
- icon
- title

It would be awesome if Tauri could support some of the more advanced features that [electron notifications](https://www.electronjs.org/docs/latest/api/notification#new-notificationoptions) offers. In particular the `actions` for MacOS would be really game-changing.

I know there is a ton of stuff already being worked on but was curious if there was anything on the roadmap for Notifications coming up? Thanks!

Some notification features in particular that are currently available on Electron but would be awesome to have on Tauri include: 

```yaml
hasReply boolean (optional) macOS - Whether or not to add an inline reply option to the notification.

replyPlaceholder string (optional) macOS - The placeholder to write in the inline reply input field.

# This one would be really awesome
actions [NotificationAction[]](https://www.electronjs.org/docs/latest/api/structures/notification-action) (optional) macOS - Actions to add to the notification. Please read the available actions and limitations in the NotificationAction documentation.

closeButtonText string (optional) macOS - A custom title for the close button of an alert. An empty string will cause the default localized text to be used.

toastXml string (optional) Windows - A custom description of the Notification on Windows superseding all properties above. Provides full customization of design and behavior of the notification.
```

**Comment by FabianLars** [comment]
> if there was anything on the roadmap for Notifications coming up?

Not really, no. I mean, we'd like to see it improved but we _currently_ have no resources to spend on https://github.com/hoodie/notify-rust ourselves, especially macos specific stuff is always tricky :/

**Reply by chiqors** [reply]
That's unfortunate, this is what keeps me using electron. Hopefully there will be some good news in future
